File: clients/mobile/lib/utils/workflow_state_util.dart
import '../services/i18n_service.dart';

/// Maps workflow `current_state` codes to localized labels; falls back to raw code.
String workflowStateLabel(String state) {
  final trimmed = state.trim();
  if (trimmed.isEmpty) {
    return EmcapLocale.t('common.emptyValue');
  }
  final key = 'platform.workflow.state.$trimmed';
  final label = EmcapLocale.t(key);
  return label == key ? trimmed : label;
}

File: clients/web/src/app/shared/utils/workflow-state.util.ts
import type { I18nService } from '../services/i18n.service';

/** Maps workflow `current_state` codes to localized labels; falls back to raw code. */
export function workflowStateLabel(state: string, i18n: I18nService): string {
  const trimmed = state.trim();
  if (!trimmed) {
    return i18n.t('common.emptyValue');
  }
  const key = `platform.workflow.state.${trimmed}`;
  const label = i18n.t(key);
  return label === key ? trimmed : label;
}
